octopoes/octopoes/models/ooi/dns/records.py

```diff
@@ -185,3 +185,43 @@ class DNSCAARecord(DNSRecord):
     # without interior spaces or (2) a quoted string.
     value: str
     _natural_key_attrs = ["hostname", "flags", "tag", "value"]
+
+
+class DNSGPOSRecord(DNSRecord):
+    # RFC 1712
+    object_type: Literal["DNSGPOSRecord"] = "DNSGPOSRecord"
+    dns_record_type: Literal["GPOS"] = "GPOS"
+
+    latitude: str | None = None
+    longitude: str | None = None
+    altitude: str | None = None
+
+    @property
+    def natural_key(self) -> str:
+        sha = hashlib.sha1(self.value.encode("UTF-8")).hexdigest()
+        key = super().natural_key
+        return key.replace(self.value, sha)
+
+    _reverse_relation_names = {"hostname": "dns_gpos_records"}
+
+
+class DNSLOCRecord(DNSRecord):
```

Comment:
Personally I would extend from `DNSGPOSRecord`, since it's a more limited early version of the LOC record and share the basic properties (latitude, longitude, altitude)

Reply:
Makes sense (@underdarknl also initially implemented it that way), but extending from OOI that are actual types is not possible I think. So extending from DNSRecord is possible since DNSRecord is not an actual type, but extending  DNSPOSRecord is not possible. The problem seems to be the object_type attribute...?